### ISSUE TITLE
Fails to parse Postgres range operators

### ISSUE THREAD

**Issue by loafofpiecrust** [sqlfluff/sqlfluff#2692]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I tried to use sqlfluff on a SQL query that deals with range columns in Postgres. It uses the `&&` and `&>` [range operators](https://www.postgresql.org/docs/current/functions-range.html), but `sqlfluff lint` failed to parse their usage.

### Expected Behaviour

I expect sqlfluff to support Postgres range operators, especially since range types were introduced back in Postgres 9.

### Observed Behaviour

SQLFluff fails to parse range operators in a postgres dialect SQL file, preventing it from fixing lint issues in the file. The query works just fine on my Postgresql v14 server. Relevant where clause of my SQL query:

```sql
WHERE word.character_range && paragraph.character_range
  AND word.character_range &> paragraph.character_range
```

SQLFluff output:
```sh
❯ sqlfluff lint types/queries/words_in_paragraph.sql
== [types/queries/words_in_paragraph.sql] FAIL                                                        
L:   9 | P:  27 |  PRS | Line 9, Position 27: Found unparsable section: ' &&
                       | paragraph.character_range\n AND word...'
All Finished 📜 🎉!
```

### How to reproduce

To be explicit, here's the full SQL file I'm trying to lint as `words_in_paragraph.sql`:

```sql
SELECT
  word.*,
  paragraph.id AS paragraph_id
FROM word
INNER JOIN paragraph ON paragraph.page_id = word.page_id
WHERE paragraph.id = ANY($1)
  AND word.character_range IS NOT NULL
  AND word.character_range && paragraph.character_range
  AND word.character_range &> paragraph.character_range
```

I ran `sqlfluff lint words_in_paragraph.sql` and got the output above.

### Dialect

postgres

### Version

```
sqlfluff, version 0.10.0
```

```
Python 3.9.10
```

### Configuration

```ini
[sqlfluff]
dialect = postgres
templater = placeholder

[sqlfluff:rules]
tab_space_size = 2

[sqlfluff:templater:placeholder]
param_style = numeric_dollar
1 = 'one'
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
